### PR TITLE
Fix pedersen padding

### PR DESCRIPTION
**Pull request by rozbb**
## Description

This fixes the Pedersen padding bug #76. To do this, I simply pad the input _bits_ rather than _bytes_, so that the padding length check succeeds. I've also changed the tests slightly so that they're regression tests. I wasn't sure the best way to do it so let me know if you'd rather the parameters be different. To demonstrate the bug, you just need `NUM_WINDOWS * WINDOW_SIZE` to not be divisible by 8.

closes: #76 

---

Before we can merge this PR, please make sure that all the following items have been
checked off. If any of the checklist items are not applicable, please leave them but
write a little note why.

- [X] Targeted PR against correct branch (main)
- [X] Linked to Github issue with discussion and accepted design OR have an explanation in the PR that describes this work.
- [X] Wrote unit tests
- [X] Updated relevant documentation in the code
- [X] Added a relevant changelog entry to the `Pending` section in `CHANGELOG.md`
- [X] Re-reviewed `Files changed` in the Github PR explorer

### CODE REVIEW

File: src/commitment/pedersen/constraints.rs [hunk 1]
```diff
@@ -62,28 +62,26 @@ where
...
-        // Allocate new variable for commitment output.
-        let input_in_bits: Vec<Boolean<_>> = padded_input
+        // Convert input bytes to little-endian bits
+        let mut input_in_bits: Vec<Boolean<_>> = input
             .iter()
             .flat_map(|byte| byte.to_bits_le().unwrap())
             .collect();
+
+        // Pad input to `W::WINDOW_SIZE * W::NUM_WINDOWS`.
+        let padding_size = (W::WINDOW_SIZE * W::NUM_WINDOWS) - input_in_bits.len();
```
Comment (Pratyush) [inline src/commitment/pedersen/constraints.rs:72]: maybe we should make this a `checked_sub`, and error if `input_in_bits.len() > W::WINDOW_SIZE * W::NUM_WINDOWS`?

**Comment by rozbb** [comment]
That is already checked in the assert at the top of the function. Should this be in addition or in the place of?

**Review by Pratyush — APPROVED** [review]

**Comment by Pratyush** [comment]
Oops, good point.

File: CHANGELOG.md [hunk 2]
```diff
@@ -3,6 +3,7 @@
 ## Pending
 
 - Implemented SHA-256 CRH
+- Fixed Pedersen padding bug [\#76](https://github.com/arkworks-rs/crypto-primitives/issues/76)
```
Comment (Pratyush) [inline CHANGELOG.md:6]: Small nit, but could you change the format to match the other entries? So moving the following entries to the `Features` section?

```suggestion
- [\#77](https://github.com/arkworks-rs/crypto-primitives/pull/77) Implement SHA-256 CRH.
- [\#76](https://github.com/arkworks-rs/crypto-primitives/pull/79) Fix Pedersen padding bug.
```

**Pull request merged**